/*
 *  Copyright (C) 2023 â€“ 2024 Advanced Micro Devices, Inc. All rights reserved.
 *  Licensed under the MIT License.
 */

syntax = "proto3";
package vaip_core;
message AnchorPointProto {
  string name = 1;
  string op_type = 2;
  oneof operand {
    string origin_node = 3;
    AnchorPointProto next = 4;
  }
  string pass = 5;
  AnchorPointAttributeProto attribute = 6;
}

message AnchorPointAttributeProto {
  oneof op_attr {
    bytes unknown_attr = 104;
    AnchorPointTransposeOpAttr transpose_attr = 106; //
    AnchorPointPadOpAttr pad_attr = 107; //
    AnchorPointFixAttr fix_attr = 110; //
    AnchorPointQdqAttr qdq_attr = 111; //
  }
}
message AnchorPointTransposeOpAttr{
  repeated int64 order = 1;
}

message AnchorPointPadOpAttr{
  // TODO, add document here.
  repeated int64 paddings = 1;
}

message AnchorPointFixAttr {
  int64 fix_point = 1;
}

message AnchorPointQdqAttr {
  float scale = 1;
  int64 zero_point = 2;
}
